src/crypto-providers/signingValidation.ts

```diff
@@ -0,0 +1,232 @@
+import { Errors } from '../errors'
+import {
+  StakeCredentialType,
+  TxCertificateKeys,
+  _UnsignedTxParsed,
+} from '../transaction/types'
+import {
+  SigningMode,
+  SigningParameters,
+} from '../types'
+import {
+  certificatesWithStakeCredentials,
+  filterSigningFiles,
+} from './util'
+
+const _validateNonMultisigTx = (unsignedTxParsed: _UnsignedTxParsed) => {
```

Comment:
Per Jakub's suggestion above, I abstracted the function content away into `_validateStakeCredentialsAreAllOfType`, so that we can use it for `validateOrdinaryTx` too. https://github.com/vacuumlabs/cardano-hw-cli/pull/108/commits/35fc2a62f021a63ab9731ab5cb039fdf34b76bc8